commands/deploy_test.go

```diff
@@ -0,0 +1,50 @@
+package commands
+
+import (
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCommandsDeployBuildFilelistIgnoresFiles(t *testing.T) {
+	assert := assert.New(t)
+
+	// Setup
+	dir := filepath.Join("testdata", "deploy", "tree")
+	ignores := []string{".git", "foo"}
+
+	// Invoke
+	paths, err := BuildFilelist(dir, ignores)
+
+	// Test
+	assert.NoError(err)
+	assert.Greater(len(paths), 0)
+	for _, p := range paths {
+		for _, i := range ignores {
+			assert.NotContains(p, i)
+		}
+	}
+}
+
+func TestCommandsDeployBuildFilelistErrorsOnNonExistentDirectory(t *testing.T) {
+	assert := assert.New(t)
+
+	// Setup
+	testCases := []string{
+		filepath.Join("testdata", "deploy", "non-existent-tree"),
+		filepath.Join("testdata", "deploy", "file"),
+	}
+	ignores := []string{}
```

Comment:
(nitpick) I prefer the default values:
`var ignores []string`

Reply:
Good pickup, I prefer that syntax too. 

Fixed in 9a32807.